conans/test/integration/toolchains/env/test_virtualenv_object_access.py

```diff
@@ -0,0 +1,68 @@
+import os
+import platform
+import textwrap
+
+import pytest
+
+from conans.client.tools.env import _environment_add
+from conans.test.assets.genconanfile import GenConanfile
+from conans.test.utils.tools import TestClient
+from conans.tools import save
+
+
+@pytest.fixture
+def client():
+    client = TestClient()
+    conanfile = str(GenConanfile())
+    conanfile += """
+    def package_info(self):
+        self.buildenv_info.define("Foo", "MyVar!")
+        self.runenv_info.define("runFoo", "Value!")
+
+        self.buildenv_info.append("Hello", "MyHelloValue!")
+    """
+    client.save({"conanfile.py": conanfile})
+    client.run("create . foo/1.0@")
+    save(client.cache.new_config_path, "tools.env.virtualenv:auto_use=True")
+    return client
+
+
+def test_virtualenv_object_access(client):
+    conanfile = textwrap.dedent("""
+    import os
+    from conans import ConanFile
+    from conan.tools.env import VirtualEnv
+
+    class ConanFileToolsTest(ConanFile):
+        requires = "foo/1.0"
+
+        def build(self):
+          env = VirtualEnv(self)
+          build_env = env.build_environment().to_dict()
```

Comment:
Everything looks good, just a small thing, this ``to_dict()`` doesn't read great to me. Maybe is just the name? Or should we create a dict-like access to ``Environment`` class?

In any case, probably just personal opinion, so I am good with this.

Reply:
updated